File: src/main/java/com/patres/timetable/repository/IntervalRepository.java
package com.patres.timetable.repository;

import com.patres.timetable.domain.Interval;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Repository;

import org.springframework.data.jpa.repository.*;

import java.util.Set;


/**
 * Spring Data JPA repository for the Interval entity.
 */
@SuppressWarnings("unused")
@Repository
public interface IntervalRepository extends JpaRepository<Interval,Long> {

    Page<Interval> findByPeriodId(Pageable pageable, Long periodId);
}

File: src/main/kotlin/com/patres/timetable/repository/IntervalRepository.kt
package com.patres.timetable.repository

import com.patres.timetable.domain.Interval
import com.patres.timetable.domain.Period
import com.patres.timetable.domain.Teacher
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository

import org.springframework.data.jpa.repository.*

@Repository
interface IntervalRepository : JpaRepository<Interval, Long> {

    fun findByPeriodId(periodId: Long): Set<Interval>

    fun findFirstByPeriodIdAndIncludedStateTrueOrderByStartDate(periodId: Long): Interval

}
